modules/apps/headless/headless-batch-engine/headless-batch-engine-test/src/testIntegration/java/com/liferay/headless/batch/engine/resource/v1_0/test/ExportTaskResourcePerformanceTest.java

```diff
@@ -0,0 +1,143 @@
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Vendel Toreki
+ */
+@RunWith(Arquillian.class)
+public class ExportTaskResourcePerformanceTest
+	extends BaseTaskResourcePerformanceTestCase {
+
+	@Test
+	public void testPostExportTaskWithUserAccount() throws Exception {
+		_generateTestUserAccounts(recordsCount);
+		_testPostExportTask(
+			"com.liferay.headless.admin.user.dto.v1_0.UserAccount");
+	}
+
+	private void _generateTestUserAccounts(int count) throws Exception {
+		for (int i = 0; i < count; ++i) {
+			UserTestUtil.addUser();
+		}
+	}
+
+	private void _testPostExportTask(String className) throws Exception {
+		if (_log.isInfoEnabled()) {
+			_log.info("ClassName: " + className);
+		}
+
+		Map<String, String> classNamePartsMap = splitClassName(className);
+
+		try (Closeable closeable = startTimer()) {
+			HttpInvoker httpInvoker = HttpInvoker.newHttpInvoker();
+
+			httpInvoker.header(
+				HttpHeaders.ACCEPT, ContentTypes.APPLICATION_JSON);
+			httpInvoker.userNameAndPassword("test@liferay.com:test");
+			httpInvoker.httpMethod(HttpInvoker.HttpMethod.POST);
+
+			StringBundler sb = new StringBundler(
+				classNamePartsMap.containsKey("taskItemDelegateName") ? 6 : 4);
+
+			sb.append("http://localhost:8080/o/headless-batch-engine/v1.0");
+			sb.append("/export-task/");
+			sb.append(classNamePartsMap.get("className"));
+			sb.append("/JSON");
+
+			if (classNamePartsMap.containsKey("taskItemDelegateName")) {
+				sb.append("?taskItemDelegateName=");
+				sb.append(classNamePartsMap.get("taskItemDelegateName"));
+			}
+
+			httpInvoker.path(sb.toString());
+
+			HttpInvoker.HttpResponse response = httpInvoker.invoke();
+
+			ExportTask exportTask = ExportTaskSerDes.toDTO(
+				response.getContent());
+
+			String externalReferenceCode =
+				exportTask.getExternalReferenceCode();
+
+			while (true) {
+				exportTask = ExportTaskSerDes.toDTO(
+					getHttpResponseContent(
+						"http://localhost:8080/o/headless-batch-engine/v1.0" +
+							"/export-task/by-external-reference-code/" +
+								externalReferenceCode));
+
+				if (Objects.equals(
+						exportTask.getExecuteStatusAsString(), "COMPLETED")) {
+
+					break;
+				}
+				else if (Objects.equals(
+							exportTask.getExecuteStatusAsString(), "FAILED")) {
+
+					throw new AssertionError(exportTask.getErrorMessage());
+				}
+			}
+
+			httpInvoker = HttpInvoker.newHttpInvoker();
+
+			httpInvoker.header(
+				HttpHeaders.ACCEPT, ContentTypes.APPLICATION_OCTET_STREAM);
+			httpInvoker.userNameAndPassword("test@liferay.com:test");
+			httpInvoker.httpMethod(HttpInvoker.HttpMethod.GET);
+
+			httpInvoker.path(
+				StringBundler.concat(
+					"http://localhost:8080/o/headless-batch-engine/v1.0",
+					"/export-task/by-external-reference-code/",
+					externalReferenceCode, "/content"));
+
+			HttpInvoker.HttpResponse httpResponse = httpInvoker.invoke();
+
+			try (InputStream inputStream = new UnsyncByteArrayInputStream(
```

Comment:
Should we consider splitting the test into two different ones? One would be with generating the exported data and the other one would be downloading the content

Reply:
Makes sense. Depends on the requirements, whether we want to see these values separated. Now it even includes the ZIP extract time.

Reply:
What do you think @4lejandrito ? Should we discuss this with PM?

Reply:
Can't we keep a single test and measure both steps independently? This way we can reuse the data. We already split between export and import task resource, and the drawback is that data cannot be reused and we have to duplicate effort.